src/hashset.c

```diff
@@ -1260,10 +1273,13 @@ size_t hashsetScan(hashset *s, size_t cursor, hashsetScanFunction fn, void *priv
      * indicate that the full scan is completed. */
     int cursor_passed_zero = 0;
 
-    /* Mask the start cursor to the bigger of the tables, so we can detect if we
+    /* Mask the start cursor to the smaller of the tables, so we can detect if we
      * come back to the start cursor and break the loop. It can happen if enough
      * tombstones (in both tables while rehashing) make us continue scanning. */
-    cursor = cursor & (expToMask(s->bucket_exp[0]) | expToMask(s->bucket_exp[1]));
+    cursor &= expToMask(s->bucket_exp[0]);
+    if (hashsetIsRehashing(s)) {
+        cursor &= expToMask(s->bucket_exp[1]);
+    }
```

Comment:
With this change, I guess there's a small increased risk of returning duplicates.

Time 1:

* Table size: 16 buckets.
* Scan order: 0 -> 8 -> 4 -> 12 -> 2 -> 10 -> 6 -> 14 -> 1 -> ... -> 15. 
* `SCAN 0` returned new cursor 8.

Time 2:
* Table sizes 16 and 2. Shrinking started. Scan order: 0 -> 1.
* User continues previous scan with `SCAN 8`.
* Old implementation: `SCAN 8` returns elements from small table bucket 0 (masked with small table mask) and large table bucket 8 (masked with large table mask). Large table bucket 0 can safely be skipped.
* New implementation: `SCAN 8` returns elements from small table bucket 0 (masked with small table mask) and large table bucket 0 and 8 (expansion of cursor masked with small table mask).
* Both implementaions: Return new cursor 1.

Reply:
Yes, maybe. But it doesn't matter much, especially if we want to change probing to chaining anyway... Let's not spend too much effort on this.